### ISSUE TITLE
This package added 400ms to startup time.

### ISSUE THREAD

**Issue by granteagon** [baldurh/atom-status-bar-blame#8]
Have gotten this message several times when I look at the settings page.  My editor has been slow lately so I'm trying to hunt down every possible issue.

1. I have a brand new machine with 32 gigs of ram.
2. I only have a handful of packages (this is one is unfortunately the slowest).
3. I'm not running a lot of other stuff on the machine.
4. I have two panes open in the project I'm working on with about 10 files in total open, all files are less than 100 lines of code.

Let me know if you need anymore info. I enjoy the package and don't want to disable it if I don't have to.

**Comment by baldurh** [comment]
Interesting. Thanks for posting.

Is it consistently 400ms? I get between 90-105ms in a project with either one or two panes and a lot of big open files in tabs. The number of files shouldn’t matter because it only checks the currently focused TextEditor at any given time.

But it’s still too much I think. I know there are some speed optimisations that can be made both in regards of startup time and in general. I’ve already started working on it. I hope I will be able to finish it soon 😄

**Comment by granteagon** [comment]
Consistent isn't the word, but definitely every other launch or so.  Let me know if I can help with the work.

**Comment by baldurh** [comment]
I’ve stripped out two dependencies and I’m now getting ~22ms startup time. One of those was causing the most problems and I believe it’s because it was using Bluebird.
I created PR #10 If you’re interested in trying this out or looking at/working on the code 😃

I haven’t decided yet if there are any further changes needed at this time. I’m open to suggestions 😉 

Btw, I noticed that the first startup time after package code change is way higher than the following startups. Not sure if that is also the case after updating packages. Just keep that in mind if you decide to check out the code and do and `apm link` to test this 👍
